sql/parser/encode.go

```diff
@@ -38,16 +38,37 @@ var (
 )
 
 func encodeSQLString(buf []byte, in []byte) []byte {
+	// Changed if escape sequences necessary. See:
+	// http://www.postgresql.org/docs/9.0/static/sql-syntax-lexical.html
+	var strMod byte
+	for _, ch := range in {
+		if encodedChar := encodeMap[ch]; encodedChar != dontEscape {
+			// If a character which we want to escape is found, update strMod.
+			strMod = 'e' // TODO(tschottdorf) should this be 'E'?
+			break
+		}
+	}
```

Comment:
`\` isn't a character that needs escaping, except when in a string that needs escaping. Unless backtracking through the already written buffer is worth it, I think you just need a previous single pass through. Or were you talking about something else?